cmd/fleetctl/apple_mdm.go

```diff
@@ -44,124 +52,137 @@ func appleMDMCommand() *cli.Command {
+
+			err = os.WriteFile(apnsKeyPath, apnsKeyPEM, 0600)
+			if err != nil {
+				return fmt.Errorf("write private key: %w", err)
 			}
-			if err := os.WriteFile(keyPath, keyPEM, 0o600); err != nil {
-				return fmt.Errorf("write %s: %w", keyPath, err)
+
+			client := fleethttp.NewClient(fleethttp.WithTimeout(10 * time.Second))
+
+			err = apple_mdm.GetSignedAPNSCSR(client, apnsCSR)
+			if err != nil {
+				return fmt.Errorf("get signed apns csr: %w", err)
 			}
-			fmt.Printf("Successfully generated SCEP CA: %s, %s.\n", certPath, keyPath)
-			fmt.Printf("Set FLEET_MDM_APPLE_SCEP_CA_CERT_PEM=$(cat %s) FLEET_MDM_APPLE_SCEP_CA_KEY_PEM=$(cat %s) when running Fleet.\n", certPath, keyPath)
-			return nil
-		},
-	}
-}
 
-func appleMDMSetupAPNSCommand() *cli.Command {
-	return &cli.Command{
-		Name:  "apns",
-		Usage: "Commands to setup APNS certificate",
-		Subcommands: []*cli.Command{
-			appleMDMSetupAPNSInitCommand(),
-			appleMDMSetupAPNSFinalizeCommand(),
-		},
-	}
-}
+			// init scep ca
+			scepCACert, scepCAKey, err := apple_mdm.NewSCEPCACertKey()
+			if err != nil {
+				return fmt.Errorf("init scep CA: %w", err)
+			}
+
+			scepCACertPEM := apple_mdm.EncodeCertPEM(scepCACert)
+			err = os.WriteFile(scepCACertPath, scepCACertPEM, 0600)
+			if err != nil {
+				return fmt.Errorf("write scep ca certificate: %w", err)
+			}
+
+			scepCAKeyPEM := apple_mdm.EncodePrivateKeyPEM(scepCAKey)
+			err = os.WriteFile(scepCAKeyPath, scepCAKeyPEM, 0600)
+			if err != nil {
+				return fmt.Errorf("write scep ca private key: %w", err)
+			}
+
+			fmt.Fprintf(
+				c.App.Writer,
+				`Success!
+
+Generated your APNs key at %s
+
+Generated your SCEP certificate at %s
+
+Generated your SCEP key at %s
+
+Go to your email to download a CSR from Fleet. Then, visit https://identity.apple.com/pushcert to upload the CSR. You should receive an APNs certificate in return from Apple.
+
+Next, use the generated certificates to deploy Fleet with `+"`mdm`"+` configuration: https://fleetdm.com/docs/deploying/configuration#mdm
+`,
+				apnsKeyPath,
+				scepCACertPath,
+				scepCAKeyPath,
+			)
 
-func appleMDMSetupAPNSInitCommand() *cli.Command {
-	return &cli.Command{
-		Name:  "init",
-		Usage: "Start APNS certificate configuration",
-		Action: func(c *cli.Context) error {
-			// TODO(lucas): Implement command.
-			fmt.Println("Not implemented yet.")
 			return nil
 		},
 	}
 }
 
-func appleMDMSetupAPNSFinalizeCommand() *cli.Command {
-	var encryptedReq string
+func appleMDMSetupCommand() *cli.Command {
 	return &cli.Command{
-		Name:  "finalize",
-		Usage: "Finalize APNS certificate configuration",
-		Flags: []cli.Flag{
-			&cli.StringFlag{
-				Name:        "encrypted-req",
-				Usage:       "File path of the encrypted .req p7m file",
-				Destination: &encryptedReq,
-				Required:    true,
-			},
-		},
-		Action: func(c *cli.Context) error {
-			// TODO(lucas): Implement command.
-			fmt.Println("Not implemented yet.")
-			return nil
+		Name:  "setup",
```

Comment:
same as above:

```suggestion
		Name:  "setup",
		Hidden:  true, // TODO: temporary, until the MDM feature is officially released
```

Reply:
This command will eventually be removed. Since we are already setting the hidden field on the parent command, I don't think we need to set it everywhere.

Reply:
sounds good to me, marking as solved.